### ISSUE TITLE
File exists path empty when creating new notebook

### ISSUE THREAD

**Issue by AlecAivazis** [jupyter/help#20]
Hey all,

Sorry to keep flooding your help channels but I have run into another issue for which I could use some advice. 

When creating a new notebook, the `file_exists` handler is called (I'm assuming to check for the conflicts with the name of the notebook its about to create). However, my function is passed an empty string for `path` so I can't really check if there is already a notebook with the name. I saw on the documentation a line that says `The server is responsible for naming the new file` but I dont really see where that happens or how I specify the new name/path.

**Comment by minrk** [comment]
If you get `''` passed to file_exists, you should be able to return False, because there is no file at the root.

**Comment by AlecAivazis** [comment]
My function is passed `''` when creating a new file in a directory that already has notebooks. This occurs before the call to `save` so I feel like I'm missing a step when the server figures out the next name (ie Untitled1 -> Untitled2) that is preventing it from doing `''`->Untitled1.ipynb. Does that sound reasonable? If so, could you point me to where that happens?

**Comment by minrk** [comment]
`new_untitled` is [here](https://github.com/jupyter/notebook/blob/4.1.0/notebook/services/contents/manager.py#L302). It's not obvious why `file_exists('')`  is getting called, but the return value shouldn't be an issue. There will never be a file there, for any contents manager.

**Comment by AlecAivazis** [comment]
So unless i'm mistaken (which is likely), `new_untitled` receives the current path shown by the explorer (ie, i the reason i'm recieving `''` is because i'm adding the file to root). Problem is, I dont see where the logic that determines the new value for the filename is. It looks like `increment_filename` is called but since it's always passed `''`, I always recieve `Untitled1.ipynb`.

**Comment by AlecAivazis** [comment]
So after awhile, I figured out that the problem was due to me misunderstanding the semantics behind `file_exists`. Since I am making a restful service to manage the content, I have separate endpoints for files and noteboooks (not sure if that's the right thing to do, but it's what i'm doing currently) and I assumed that `file_exists` always needs to hit the `file` endpoint, but it actually needs to hit the endpoint according to the `type` of the model. Thanks for all your help @minrk - you're the best!
